### PR TITLE
[issue-146] Throw on api / env network mismatch

### LINKED ISSUE #146: Validate CARDANO_NETWORK env matches Cardano.networkName otherwise throw
In the NetworkInfo feature a mismatch here is an undesirable state. Retaining the configured setting allows for application bundling optimisations.

### PR DESCRIPTION
**Pull request by DominikGuzei**
Fixes #146 by throwing an error if the api and environment network are not matching.

### CODE REVIEW

**Comment by DominikGuzei** [comment]
@rhyslbw this breaks `yarn dev` locally because the docker API provides `mainnet` as `cardanoNetwork` value

File: next.config.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ require('dotenv').config();
 const webpack = require('webpack');
 
 const resourcesDir = path.join(__dirname, 'source/styles/resources');
-const theme = process.env.CARDANO_NETWORK || 'incentivized-testnet';
+const theme = process.env.CARDANO_NETWORK || 'mainnet';
```
Comment (rhyslbw) [inline next.config.js]: :+1:

File: source/features/network-info/store.ts [hunk 2]
```diff
@@ -85,6 +85,11 @@ export class NetworkInfoStore extends Store {
     const result = await this.networkInfoApi.fetchStatic.execute({});
     if (result) {
       const { cardano } = result;
+      if (cardano.networkName !== environment.CARDANO.NETWORK) {
+        throw new Error(
+          `API network '${cardano.networkName}' does not match environment network '${environment.CARDANO.NETWORK}'`
```
Comment (rhyslbw) [inline source/features/network-info/store.ts]:
```suggestion
          `Cardano GraphQL is connected to" '${cardano.networkName}', whereas the web app is expecting '${environment.CARDANO.NETWORK}'`. The instance of Cardano GraphQL needs to be configured to match our expected environment. 
```

Comment (rhyslbw) [reply source/features/network-info/store.ts]: Was going to suggest a custom error, but I feel a better approach is to remove the local config and make the app more network agnostic. So a quick error here is fine

**Comment by DominikGuzei** [comment]
@rhyslbw does this mean you want to merge this PR or should we switch to a more dynamic way of determining the network? This would require the refactoring to css variables first (as currently the CSS theme is baked-in during at build time)

**Comment by rhyslbw** [comment]
@DominikGuzei Let's merge this with my suggested edit to the error wording. Moving to a more dynamic design needs further consideration, and it's not required at this point

**Pull request merged**